dc-cudami-editor/src/components/modals/ImageAdderModal.jsx

```diff
@@ -0,0 +1,161 @@
+      readOnly: false,
+      step: 'initial',
+    }
+    subscribe('editor.show-image-modal', () => {
+      this.props.onToggle()
+    })
+  }
+
+  addImageToEditor = () => {
+    this.toggle()
+    const filteredAttributes = Object.entries(this.state.attributes).reduce(
+      (filteredAttributes, attribute) => {
+        if (attribute[1]) {
+          filteredAttributes[attribute[0]] = attribute[1]
+        }
+        return filteredAttributes
+      },
+      {}
+    )
+    publish('editor.add-image', filteredAttributes)
+  }
+
+  getStepComponent = t => {
+    const STEP_COMPONENT_MAPPING = {
+      initial: (
+        <ListGroup>
+          <ListGroupItem
+            className="text-decoration-none"
+            href="#"
+            onClick={() => this.setState({step: 'url'})}
+            tag="a"
+          >
+            <FontAwesomeIcon icon={faCode} /> URL
+          </ListGroupItem>
+          <ListGroupItem
+            className="text-decoration-none"
+            href="#"
+            onClick={() => this.setState({step: 'upload'})}
+            tag="a"
+          >
+            <FontAwesomeIcon icon={faUpload} /> Upload
+          </ListGroupItem>
+        </ListGroup>
+      ),
+      metadata: (
+        <ImageMetadataForm
+          attributes={this.state.attributes}
+          onChange={(name, value) => this.setAttribute(name, value)}
+          onSubmit={() => this.setState({step: 'renderingHints'})}
+        />
+      ),
+      renderingHints: (
+        <ImageRenderingHintsForm
+          attributes={this.state.attributes}
+          onChange={(name, value) => this.setAttribute(name, value)}
+          onSubmit={() => this.addImageToEditor()}
+        />
+      ),
+      upload: (
+        <ImageUploadForm
+          apiContextPath={this.props.apiContextPath}
+          onChange={attributes => this.setAttributes(attributes)}
+          onSubmit={() => this.setState({readOnly: true, step: 'url'})}
+        />
+      ),
+      url: (
+        <ImageUrlForm
+          attributes={this.state.attributes}
+          onChange={(name, value) => this.setAttribute(name, value)}
+          onSubmit={() => this.setState({step: 'metadata'})}
+          readOnly={this.state.readOnly}
+        />
+      ),
+    }
+    return STEP_COMPONENT_MAPPING[this.state.step]
+  }
+
+  getStepTitle = t => {
+    const STEP_TITLE_MAPPING = {
+      initial: 'Bild einfügen über',
+      renderingHints: 'Darstellung',
+      upload: 'Hochladen',
+      url: 'URL eingeben',
+      metadata: 'Metadaten eingeben',
+    }
+    return STEP_TITLE_MAPPING[this.state.step]
+  }
+
+  setAttribute = (name, value) => {
+    this.setState({
+      attributes: {
+        ...this.state.attributes,
+        [name]: value,
+      },
+    })
+  }
+
+  setAttributes = attributes => {
+    this.setState({
+      attributes: {
+        ...this.state.attributes,
+        ...attributes,
+      },
+    })
+  }
+
+  toggle = () => {
```

Comment:
Der Name sagt `toggle`, aber was die Funktion macht ist ein `reset`, vielleicht wäre hier ein Kommentar ganz gut der erklärt dass reset == verstecken. `toggle` impliziert auch immer dass es in beide Richtungen geht (an => aus, aus => an), das ist hier ja nciht der fall, oder? Vllt eher `hide`, `destroy`, `reset` o.Ä. als Name?

Reply:
Wie wär's mit `hideAndReset`?

Reply:
Die Funktion heißt jetzt `destroy`.